cmd/av/branch.go

```diff
@@ -167,6 +198,111 @@ internal tracking metadata that defines the order of branches within a stack.`),
 	},
 }
 
+func branchSplit(repo *git.Repo, currentBranchName string, newBranchName string) error {
+
+	// Get the current branch reference
+	currentBranchRefName := plumbing.NewBranchReferenceName(currentBranchName)
+	currentBranchRef, err := repo.GoGitRepo().Reference(currentBranchRefName, true)
+
+	if err != nil {
+		return fmt.Errorf("failed to get reference for current branch %s: %w", currentBranchName, err)
+	}
+
+	lastCommitHash := currentBranchRef.Hash()
+	// Get the last commit object
+	lastCommit, err := repo.GoGitRepo().CommitObject(lastCommitHash)
+	if err != nil {
+		return fmt.Errorf("failed to get last commit: %w", err)
+	}
+
+	headRef, err := repo.GoGitRepo().Head()
+	if err != nil {
+		return fmt.Errorf("failed to get HEAD reference: %v", err)
+	}
+
+	headHash := headRef.Hash()
+	if headHash != lastCommitHash {
+		return fmt.Errorf("user is not on the latest commit")
+	}
+	// Generate a branch name if none is provided
+	if newBranchName == "" {
+		newBranchName = sanitizeBranchName(lastCommit.Message)
+	}
+
+	// Get the parent of the last commit (HEAD~1)
+	parentCommitIter := lastCommit.Parents()
+	parentCommit, err := parentCommitIter.Next()
+	if err != nil {
+		return fmt.Errorf("failed to get parent commit: %w", err)
+	}
+
+	// Create a new branch pointing to the last commit
+	newBranchRefName := plumbing.NewBranchReferenceName(newBranchName)
+	newBranchRef := plumbing.NewHashReference(newBranchRefName, lastCommitHash)
+	if err := repo.GoGitRepo().Storer.SetReference(newBranchRef); err != nil {
+		return fmt.Errorf("failed to create new branch: %w", err)
+	}
+
+	updatedCurrentBranchRef := plumbing.NewHashReference(currentBranchRefName, parentCommit.Hash)
+	if err := repo.GoGitRepo().Storer.SetReference(updatedCurrentBranchRef); err != nil {
+		return fmt.Errorf("failed to update current branch: %w", err)
+	}
+	// Switch to the newly created branch
+	worktree, err := repo.GoGitRepo().Worktree()
+	if err != nil {
+		return fmt.Errorf("failed to get worktree: %w", err)
+	}
+
+	if err := worktree.Checkout(&gogit.CheckoutOptions{
+		Branch: newBranchRefName,
+		Create: false, // We're switching, not creating
+		Force:  false, // Avoid overwriting changes
+	}); err != nil {
+		return fmt.Errorf("failed to switch to new branch '%s': %w", newBranchName, err)
+	}
+
+	fmt.Fprint(
+		os.Stdout,
+		colors.Success(fmt.Sprintf("Successfully split the last commit into a new branch %s.\n", newBranchName)),
+	)
+
+	// Adopt new branch to av database
+	err = adoptCmd.RunE(nil, []string{"parent", currentBranchName})
```

Comment:
We can call `adoptForceAdoption(repo, db, newBranch, oldBranch)` directly

Reply:
Hi @tulioz , I update to use `adoptForceAdoption` for updating the av database.
Also, maybe I misunderstand what you mean by "tip of the branch". I assumed that the branch needs to be on its last commit for this split command to run. Is this correct? Because if that's correct then this should work as expected. 
When you switch to another branch then the user automatically is on the last commit of that branch, so the split should work.
But if you switch to another commit, like this:
`chafid@WINDOWS-4D6659B:~/repository/test_repo$ av switch commit-2
chafid@WINDOWS-4D6659B:~/repository/test_repo$ git log --oneline
ccf90a5 (HEAD -> commit-2) commit 2
6714a51 (commit-1) commit 1
0e32223 (origin/main, origin/HEAD, main) 1st commit
chafid@WINDOWS-4D6659B:~/repository/test_repo$ av switch 6714a51
chafid@WINDOWS-4D6659B:~/repository/test_repo$ echo "Third commit" > commit-3.txt
chafid@WINDOWS-4D6659B:~/repository/test_repo$ git add commit-3.txt
chafid@WINDOWS-4D6659B:~/repository/test_repo$ git commit -m "commit 3"
[detached HEAD e17cd79] commit 3
 1 file changed, 1 insertion(+)
 create mode 100644 commit-3.txt
chafid@WINDOWS-4D6659B:~/repository/test_repo$ av branch --split
====================================================
The split operation was aborted.
The original branch  remains intact.
`

Reply:
Hi @tulioz , I update to use `adoptForceAdoption` for updating the av database.
Also, maybe I misunderstand what you mean by "tip of the branch". I assumed that the branch needs to be on its last commit for this split command to run. Is this correct? Because if that's correct then this should work as expected. 
When you switch to another branch then the user automatically is on the last commit of that branch, so the split should work.
But if you switch to another commit, like this:

chafid@WINDOWS-4D6659B:~/repository/test_repo$ av switch commit-2
chafid@WINDOWS-4D6659B:~/repository/test_repo$ git log --oneline
ccf90a5 (HEAD -> commit-2) commit 2
6714a51 (commit-1) commit 1
0e32223 (origin/main, origin/HEAD, main) 1st commit
chafid@WINDOWS-4D6659B:~/repository/test_repo$ av switch 6714a51
chafid@WINDOWS-4D6659B:~/repository/test_repo$ echo "Third commit" > commit-3.txt
chafid@WINDOWS-4D6659B:~/repository/test_repo$ git add commit-3.txt
chafid@WINDOWS-4D6659B:~/repository/test_repo$ git commit -m "commit 3"
[detached HEAD e17cd79] commit 3
 1 file changed, 1 insertion(+)
 create mode 100644 commit-3.txt
chafid@WINDOWS-4D6659B:~/repository/test_repo$ av branch --split
The split operation was aborted.
The original branch  remains intact.

This should work as expected